runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseTest.java

```diff
@@ -424,15 +428,21 @@ public String execModule(String fileName) {
 	private String locateTool(String tool) {
 		ArrayList<String> paths = new ArrayList<String>(); // default cap is about right
 
-		String pathEnv = System.getenv("PATH");
-		if (pathEnv != null) {
-			paths.addAll(Arrays.asList(pathEnv.split(pathSep)));
-		}
+		// GOROOT should have priority if set
 		String goroot = System.getenv("GOROOT");
 		if (goroot != null) {
 			paths.add(goroot + File.separatorChar + "bin");
 		}
 
+		String pathEnv = System.getenv("PATH");
+		if (pathEnv != null) {
+			paths.addAll(Arrays.asList(pathEnv.split(pathSep)));
+		}
+
+		// OS specific default locations of binary dist as last resort
+		paths.add("/usr/local/go/bin");
+		paths.add("c:\\Go\\bin");
+
```

Comment:
This code does not construct a new `PATH` or `GOPATH`, there will be no strings `c` and `\Go\bin` or `\\Go\\bin`. The ArrayList<String> is a local variable that never hits any code-path other than File(). Potentially this will call File.exists() on `/usr/local/go/bin/go` and `c:\Go\bin\go`. Both are harmless on any platform.